src/verifier/builder_visitor.rs

```diff
@@ -270,9 +345,11 @@ impl SymbolTableBuilder<'_> {
         // create record
         let fn_rec = Record::Fn {
             name: f.name.clone(),
+            is_comp_provided: self.comp_def,
```

Comment:
You should be able to get this from the `Fn` type from the `parser` instead of creating a new member field that assumes things based on the current `SymbolTable` scope.